app/models/badge_achievement.rb

```diff
@@ -32,7 +32,7 @@ def notify_recipient
   def send_email_notification
```

Comment:
This is called in an `after_create` callback, we need to update that to be `after_create_commit` to ensure the record is in the db before we attempt to send a notification.

Reply:
Agree with @mstruve here. Although, it seems to me that this was a preexisting issue since we were enqueuing this with `perform_later` before the call to `save` from the `after_create` had actually executed 😬

Reply:
@vaidehijoshi it was not an issue with DelayedJob bc the jobs were so delayed that by the time they ran the records were always committed to the db which is why we have to now switch a bunch of these now bc Sidekiq is way faster.

Reply:
Lol, what a happy (and slow) coincidence! Thanks for clarifying 👍

Reply:
@vaidehijoshi funnily enough we never encountered this issue because Delayed Job was too slow at picking up jobs. A concurrency error that wasn't because the client was too slow to trigger it :D

Reply:
That's entirely my bad, asleep at the wheel there. Thanks for catching that!